semstr/conversion/ptb.py

```diff
@@ -0,0 +1,282 @@
+            else:
+                # if terminal
+                if (istok(stack[-1], STRING_TOKEN) and
+                        istok(stack[-2], STRING_TOKEN) and
+                        istok(stack[-3], LPAREN_TOKEN)):
+                    w = Leaf(stack[-1].value, stack[-2].value, parent_id=node_id)
+                    stack.pop()
+                    stack.pop()
+                    stack.pop()
+                    stack.append(TExpr(w, node_id=node_id))
+                    self.terminals.append((w.word, w.tag, w.edge_tag, node_id))
+                # if regular node
+                else:
+                    tx = None
+                    tail = None
+                    peers = []
+                    while not istok(stack[-1], LPAREN_TOKEN):
+                        head = stack.pop()
+                        # if it is the edge tag
+                        if istok(head, STRING_TOKEN):
+                            tx = TExpr(
+                                Symbol(head.value), first_child=tail, next_sibling=None, node_id=node_id
+                            )
+                        # if it is a TExp object
+                        else:
+                            peers.append(head)
+                            head.next_sibling = tail
+                            tail = head
+                    stack.pop()
+
+                    if tx is None:
+                        tx = TExpr(None, node_id=node_id)
+
+                    tx.child_list = peers
+                    for child in tx.children():
+                        child.parent_node = tx
+                        # if the child is a terminal
+                        if isinstance(child.head, Leaf):
+                            self.node_ids_with_children.add(child.node_id)
+                            self.pending_nodes.append((child.parent_node.node_id, str(child.head.pos), child.node_id))
+                            self.node_ids_with_children.add(child.parent_node.node_id)
+                        # if the child is a regular node
+                        else:
+                            self.pending_nodes.append((child.parent_node.node_id, str(child.head), child.node_id))
+                            self.node_ids_with_children.add(child.parent_node.node_id)
+                            # node is root
+                            if child.parent_node.node_id is None:
+                                self.node_by_id[child.node_id] = None
+                    if not stack:
+                        self.node_by_id[node_id] = None
+                        yield tx
+                    else:
+                        stack.append(tx)
+                node_id += 1
+
+
+#######
+# Utils
+#######
+
+def gensym():
+    return object()
+
+
+##################
+# Lexer
+##################
+
+
+LPAREN_TOKEN = gensym()
+RPAREN_TOKEN = gensym()
+STRING_TOKEN = gensym()
+
+class Token(object):
+    _token_ids = {LPAREN_TOKEN:"(", RPAREN_TOKEN:")", STRING_TOKEN:"STRING"}
```

Comment:
Why not just use the string itself as the value?
I mean, something like `LPAREN = "("` etc.